pkg/cluster/connection_pooler_test.go

```diff
@@ -366,9 +366,7 @@ func TestConnectionPoolerSynchronization(t *testing.T) {
 		{
 			subTest: "create if doesn't exist",
 			oldSpec: &acidv1.Postgresql{
-				Spec: acidv1.PostgresSpec{
-					ConnectionPooler: &acidv1.ConnectionPooler{},
-				},
```

Comment:
This change I don't get. Now it's the same like the first unit test. Shouldn't the operator try to create the connection pooler when it's not there - even if the old spec says so? I think this was the original idea.